From abd3fd84b1dbdcfab2aaa0db87de502badd4d2a2 Mon Sep 17 00:00:00 2001
From: Jonathan Haws <jhaws@sdl.usu.edu>
Date: Sat, 31 Dec 2016 13:55:50 -0700
Subject: [PATCH] Updating Makefile to support Yocto builds

---
 Makefile | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index b852a17..dede6dc 100644
--- a/Makefile
+++ b/Makefile
@@ -34,7 +34,7 @@ export CONFIG_BATMAN_ADV_BATMAN_V=n
 
 PWD:=$(shell pwd)
 BUILD_DIR=$(PWD)/build
-KERNELPATH ?= /lib/modules/$(shell uname -r)/build
+KERNELPATH ?= $(KERNEL_SRC)
 # sanity check: does KERNELPATH exist?
 ifeq ($(shell cd $(KERNELPATH) && pwd),)
 $(warning $(KERNELPATH) is missing, please set KERNELPATH)
@@ -93,8 +93,9 @@ clean:
 	$(RM) compat-autoconf.h*
 	$(RM) -r $(BUILD_DIR)
 
-install: config $(SOURCE_STAMP)
+modules_install: config $(SOURCE_STAMP)
 	$(MAKE) -C $(KERNELPATH) $(BUILD_FLAGS) modules_install
+	ln -svf $(shell pwd)/build/Module.symvers .
 	depmod -a
 
 config:
-- 
1.9.1

