From: Jonathan Haws <jhaws@sdl.usu.edu>
Date: Sat, 31 Dec 2016 13:55:50 -0700
Subject: [PATCH] batman-adv: Provide Module.symvers during install
---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index 08ac8f65e8f51405d4972c9b7f15d3e95c6cd626..1141df8eb68a78ba945ff432c96774080ca8640c 100644
--- a/Makefile
+++ b/Makefile
@@ -91,6 +91,7 @@ clean:
 
 install: config
 	$(MAKE) -C $(KERNELPATH) $(BUILD_FLAGS) modules_install
+	ln -svf $(shell pwd)/build/Module.symvers .
 	$(DEPMOD)
 
 config:
