From: Sven Eckelmann <sven@narfation.org>
Date: Wed, 5 Dec 2018 14:17:13 +0100
Subject: [PATCH] batman-adv: Fix timer_setup on 4.12.28

Signed-off-by: Sven Eckelmann <sven@narfation.org>
---
 compat-include/linux/timer.h | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/compat-include/linux/timer.h b/compat-include/linux/timer.h
index 262c38617488e32ab53bd31d7da9249912c84ac7..2bdfcf64b90aa75e90e70773c89689a4a27f0a72 100644
--- a/compat-include/linux/timer.h
+++ b/compat-include/linux/timer.h
@@ -30,14 +30,16 @@
 #define TIMER_DATA_TYPE		unsigned long
 #define TIMER_FUNC_TYPE		void (*)(TIMER_DATA_TYPE)
 
-static inline void timer_setup(struct timer_list *timer,
-			       void (*callback)(struct timer_list *),
-			       unsigned int flags)
+static inline void batadv_timer_setup(struct timer_list *timer,
+				      void (*callback)(struct timer_list *),
+				      unsigned int flags)
 {
 	__setup_timer(timer, (TIMER_FUNC_TYPE)callback,
 		      (TIMER_DATA_TYPE)timer, flags);
 }
 
+#define timer_setup batadv_timer_setup
+
 #define from_timer(var, callback_timer, timer_fieldname) \
 	container_of(callback_timer, typeof(*var), timer_fieldname)
 
