--- a/Makefile
+++ b/Makefile
@@ -29,9 +29,9 @@
 MANPAGE = man/batctl.8
 
 # batctl flags and options
-CFLAGS += -Wall -W -std=gnu99 -fno-strict-aliasing -MD -MP
-CPPFLAGS += -D_GNU_SOURCE
-LDLIBS += -lm
+BC_CFLAGS += -Wall -W -std=gnu99 -fno-strict-aliasing -MD -MP
+BC_CPPFLAGS += -D_GNU_SOURCE
+BC_LDLIBS += -lm
 
 # disable verbose output
 ifneq ($(findstring $(MAKEFLAGS),s),s)
@@ -58,21 +58,21 @@
   LIBNL_CFLAGS += $(shell $(PKG_CONFIG) --cflags $(LIBNL_NAME))
   LIBNL_LDLIBS +=  $(shell $(PKG_CONFIG) --libs $(LIBNL_NAME))
 endif
-CFLAGS += $(LIBNL_CFLAGS)
-LDLIBS += $(LIBNL_LDLIBS)
+BC_CFLAGS += $(LIBNL_CFLAGS)
+BC_LDLIBS += $(LIBNL_LDLIBS)
 
 # standard build tools
 ifeq ($(CONFIG_BATCTL_BISECT),y)
 OBJ += $(OBJ_BISECT)
-CPPFLAGS += -DBATCTL_BISECT
+BC_CPPFLAGS += -DBATCTL_BISECT
 endif
 
 CC ?= gcc
 RM ?= rm -f
 INSTALL ?= install
 MKDIR ?= mkdir -p
-COMPILE.c = $(Q_CC)$(CC) $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -c
-LINK.o = $(Q_LD)$(CC) $(CFLAGS) $(LDFLAGS) $(TARGET_ARCH)
+COMPILE.c = $(Q_CC)$(CC) $(CFLAGS) $(BC_CFLAGS) $(CPPFLAGS) $(BC_CPPFLAGS) $(TARGET_ARCH) -c
+LINK.o = $(Q_LD)$(CC) $(CFLAGS) $(BC_FLAGS) $(LDFLAGS) $(TARGET_ARCH)
 
 # standard install paths
 PREFIX = /usr/local
@@ -96,7 +96,7 @@
 	$(COMPILE.c) -o $@ $<
 
 $(BINARY_NAME): $(OBJ)
-	$(LINK.o) $^ $(LDLIBS) -o $@
+	$(LINK.o) $^ $(LDLIBS) $(BC_LDLIBS) -o $@
 
 clean:
 	$(RM) $(BINARY_NAME) $(OBJ) $(OBJ_BISECT) $(DEP)
